### ISSUE TITLE
Error: Method addObserver must be called on the main thread

### ISSUE THREAD

**Issue by sinadalvand** [pusheco/pushe-android-studio-sample#42]
بعد از طی کردن تمام مراحل با خطا زیر در لاگ مواجه میشم:

`10-23 16:02:07.624 7387-7387/... I/Pushe: Starting Pushe initialization
10-23 16:02:07.844 7387-7433/... E/Pushe: co.pushe.plus.utils.PusheUnhandledException: Method addObserver must be called on the main thread
        at co.pushe.plus.j0.c$a.run(Schedulers.kt:4)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: java.lang.IllegalStateException: Method addObserver must be called on the main thread
        at androidx.lifecycle.LifecycleRegistry.enforceMainThreadIfNeeded(LifecycleRegistry.java:317)
        at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:172)
        at co.pushe.plus.CoreInitializer$a.subscribe(PusheInit.kt:2)
        at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
        at io.reactivex.internal.operators.observable.ObservableIgnoreElementsCompletable$IgnoreObservable.onComplete(ObservableIgnoreElementsCompletable.java:66)
        at io.reactivex.internal.operators.observable.ObservableTake$TakeObserver.onComplete(ObservableTake.java:88)
        at io.reactivex.internal.operators.observable.ObservableTake$TakeObserver.onNext(ObservableTake.java:66)
        at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)
        at io.reactivex.internal.operators.observable.ObservableDistinctUntilChanged$DistinctUntilChangedObserver.onNext(ObservableDistinctUntilChanged.java:85)
        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
        at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
        at co.pushe.plus.j0.c$a.run(Schedulers.kt:3)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288) 
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
        at java.lang.Thread.run(Thread.java:818) 
`

**Comment by sinadalvand** [comment]
Problem solved by downgrade version of "androidx.fragment:fragment-ktx" dependency to "1.2.5"

**Comment by mahdi-malv** [comment]
Thanks for the report.

I've made a fix to the latest version and published a preview (**2.2.1**).

You can check it out and see if it fixes the issue in your case:

(Published version is not currently at `JCenter` and is accessible via `preview` channel:
```
allprojects {
    repositories {
        google()
        jcenter()
        maven { url "https://dl.bintray.com/pushe/preview" }
    }
}
```

```groovy
dependencies {
   def pusheVersion = "2.2.1" // Fix is on 2.2.1
   implementation("co.pushe.plus:base:$pusheVersion")
   // ...
}
```
